src/sparseml/pytorch/sparsification/pruning/modifier_pruning_rigl.py

```diff
@@ -0,0 +1,443 @@
+    with smallest magnitude is pruned and the same
+    amount of weights are regrown according to
+    the magnitude of the gradient.
+
+    Pruner supports three sparsity strategies as in the original paper:
+    - uniform
+    - erdos_renyi
+    - erdos_renyi_kernel
+
+    Supported mask types: unstructured.
+
+    | Sample yaml:
+    |   !RigLPruningModifier
+    |       sparsity: 0.7
+    |       start_epoch: 2.0
+    |       end_epoch: 26.0
+    |       update_frequency: 4.0
+    |       params: ["re:.*weight"]
+    |       leave_enabled: True
+    |       inter_func: cubic
+    |       global_sparsity: True
+    |       mask_type: unstructured
+    |       sparsity_strategy: erdos_renyi_kernel
+    |       init_update_fraction: 0.3
+
+    :param sparsity: the final sparsity for the param to end with at end_epoch.
+        A single float for the whole model.
+    :param start_epoch: The epoch to start the modifier at
+    :param end_epoch: The epoch to end the modifier at
+    :param update_frequency: The number of epochs or fraction of epochs to update at
+        between start and end
+    :param params: A list of full parameter names or regex patterns of names to apply
+        pruning to.  Regex patterns must be specified with the prefix 're:'. __ALL__
+        will match to all parameters. __ALL_PRUNABLE__ will match to all ConvNd
+        and Linear layers' weights. If a sparsity to param mapping is defined by
+        final_sparsity, then params should be set to []
+    :param leave_enabled: True to continue masking the weights after end_epoch,
+        False to stop masking. Should be set to False if exporting the result
+        immediately after or doing some other prune
+    :param inter_func: the type of interpolation function to use:
+        [linear, cubic, inverse_cubic]
+    :param mask_type: String to define type of sparsity to apply.
+        RigL modifier supports only 'unstructured'
+    :param global_sparsity: set True to enable global pruning. If False, pruning will
+        be layer-wise. Default is True
+    :param sparsity_strategy: String to define the sparsity distribution. Following
+        the original paper one can select one of the 3 options:
+        [uniform, erdos_renyi, erdos_renyi_kernel].
+    :param init_update_fraction: The initial percentage of the weights updated -
+        pruned and regrown
+    """
+
+    _supported_masks = "unstructured"
+    _sparsity_strategies = ("uniform", "erdos_renyi", "erdos_renyi_kernel")
+
+    def __init__(
+        self,
+        sparsity: float,
+        start_epoch: float,
+        end_epoch: float,
+        update_frequency: float,
+        params: Union[str, List[str]],
+        leave_enabled: bool = True,
+        global_sparsity: bool = True,
+        mask_type: str = "unstructured",
+        sparsity_strategy: str = "erdos_renyi_kernel",
+        init_update_fraction: float = 0.3,
```

Comment:
should add `ModifierProp` properties for both of these for YAML serialization support